include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -3466,8 +3466,7 @@ namespace internal
 
   public:
     static constexpr bool value =
-      std::is_same<typename T::const_iterator,
-                   decltype(detect(std::declval<T>()))>::value;
+      !std::is_same<void, decltype(detect(std::declval<T>()))>::value;
```

Comment:
Can you update the comments in line 3404 and 3455 such that they show the return type we test for (first case) or don't (second case)?

Reply:
but those comments are NOT for this detector.

Reply:
at least `3404`. 

`3455` still applies as is. We test if  there `something ... T::begin()` where `something != void`.

Reply:
maybe I misunderstand what you would like to have in comments...

Reply:
Yes, line 3404 is not related to the pull request here, but I was too late to comment on the original pull request.
There we are saying 
```
 // ... T::local_element() const
```
but we are testing that the return type is `typename T::value_type`. What I wanted is to either be specific:
```
 // typename T::value_type T::local_element() const
```
or also test for any non-`void` return type there as well.

In line 3455, we are saying
```
 // T::const_iterator T::begin() const
```
but test for any non-`void` return type. In view of the style in line 3404, I expected

```
 // ... T::begin() const
```

Reply:
ah, ok, i will adjust the comments a bit later today